Vhar 7421 analytiikkasql

solita-staskila opened the pull request.

solita-staskila committed:
käytetään luotu-aikaleimaa, otettu materiaali taulu huomioon

solita-staskila committed:
lisätty migraatiotiedosto

solita-staskila committed:
testi materiaaleille

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_948__.sql:
```diff
@@ -11,7 +11,7 @@ BEGIN
         SELECT t.id                                                                       as toteuma_tunniste_id,
                t.sopimus                                                                  as toteuma_sopimus_id,
-               t.alkanut                                                                  as toteuma_alkanut,
+               t.luotu                                                                    as toteuma_luotu,
```
Vanhoihin migraatiotiedostoihin ei voi tehdä muutoksia.

solita-staskila replied:
Commit otettu takaisin

solita-staskila committed:
Revert "käytetään luotu-aikaleimaa, otettu materiaali taulu huomioon"

This reverts commit b37d12c229454568907784851cb4c225728c109d.

solita-staskila committed:
migraatiotiedoston nimi korjattu

maaritlaine reviewed (commented):
SQL on muutettu juuri kuten tarvitsikin, mutta tuli nyt sitten mieleen, että kun SQL nimeltä **hae-reittitoteumat-analytiikalle** hakee aikaleimojen perusteella rivit analytiikka_toteumat-taulusta, niin siellähän pitäisi myös olla hyödynnettävissä toteuma_materiaali-taulun luonti- ja muokkausaika. 

Kaivataan siis vielä iterointikierros, jolla lisätään sarakkeet analytiikka_toteumat-tauluun ja päivitetään tuon SQL:n hakuehtoa.

solita-staskila committed:
toteumat hakuun otettu materiaali-taulu huomioon

solita-staskila committed:
Revert "toteumat hakuun otettu materiaali-taulu huomioon"

This reverts commit 885ad4550730439c18f17e0ef880aa8e7f193831.

solita-staskila committed:
Merge branch 'develop' into VHAR-7421_analytiikkasql

solita-staskila committed:
lisätty analytiikka tauluun sarakkeita, muokattu kyselyitä

Analytiikka tauluun lisätty materiaalien muokkaus ja luotu ajat, mitkä lisätty myös proseduuriin jossa ne insertataan. Postgres vaati group by clauseen myös materiaali sarakkeet. Nämä sarakkeet otettu nyt huomioon hae-reittitoteumat-analytiikalle kyselyssä

rckopteri commented on src/clj/harja/kyselyt/toteumat.sql:
```diff
@@ -1473,12 +1473,18 @@ SELECT t.toteuma_tunniste_id,
 WHERE ((t.toteuma_muutostiedot_muokattu IS NOT NULL AND t.toteuma_muutostiedot_muokattu BETWEEN :alkuaika::TIMESTAMP AND :loppuaika::TIMESTAMP)
     OR (t.toteuma_muutostiedot_muokattu IS NULL AND t.toteuma_muutostiedot_luotu BETWEEN :alkuaika::TIMESTAMP AND :loppuaika::TIMESTAMP))
+
+OR ((t.toteuma_materiaali_muokattu IS NOT NULL AND t.toteuma_materiaali_muokattu BETWEEN :alkuaika::TIMESTAMP AND :loppuaika::TIMESTAMP)
```
Jotenkin ensimmäisellä katselmointikierroksella tämän sivuutin.
Minkä verran tämä hidastaa hakua? 
Tältä ainakin puuttuu indeksi, joten oletuksena haku menee todella hitaaksi.

solita-staskila replied:
nämä rivit tuli vasta sinun katselmoinnin jälkeen, ja hyvä pointti.
lisään samanlaisen indeksin mitä noille **_toteuma_muutostiedot_muokattu_** on tehty. 

voin kokeilla vielä singletonilla että sieltä tulee vastaus.

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_997__.sql:
```diff
@@ -0,0 +1,3 @@
+--Otettu analytiikka toteumien rajapinnoissa materiaalien timestampit huomioon
+ALTER TABLE analytiikka_toteumat ADD COLUMN toteuma_materiaali_luotu timestamp;
```
Lisää indeksit molemmille.

rckopteri reviewed (changes requested):
Jätin pari kommenttia suorituskyvystä

solita-staskila committed:
indeksit lisätty materiaalien timestampeille

solita-staskila committed:
Merge branch 'develop' into VHAR-7421_analytiikkasql

solita-staskila committed:
migraatio nimi
